controllers/scripts.go

```diff
@@ -236,3 +223,64 @@ func (r *IpfsReconciler) configMapScripts(
 		return nil
 	}, cmName
 }
+
+// staticAddrsFromRelayPeers Extracts all of the static addresses from the
+// given list of relayPeers.
+func staticAddrsFromRelayPeers(relayPeers []*peer.AddrInfo) ([]*ma.Multiaddr, error) {
+	relayStatic := make([]*ma.Multiaddr, 0)
+	for _, addrInfo := range relayPeers {
+		p2ppart, err := ma.NewMultiaddr("/p2p/" + addrInfo.ID.String())
+		if err != nil {
+			return nil, fmt.Errorf("could not create p2p component: %w", err)
+		}
+		for _, addr := range addrInfo.Addrs {
+			fullMa := addr.Encapsulate(p2ppart)
+			relayStatic = append(relayStatic, &fullMa)
+		}
+	}
+	return relayStatic, nil
+}
+
+// getCircuitInfo Gets address info from the list of CircuitRelays
+// and returns a list of AddrInfo.
+func (r *IpfsReconciler) getCircuitInfo(
+	ctx context.Context,
+	ipfs *clusterv1alpha1.Ipfs,
+) ([]*peer.AddrInfo, error) {
+	log := ctrllog.FromContext(ctx)
+	relayPeers := []*peer.AddrInfo{}
+	for _, relayName := range ipfs.Status.CircuitRelays {
+		relay := clusterv1alpha1.CircuitRelay{}
+		relay.Name = relayName
+		relay.Namespace = ipfs.Namespace
+		// OPTIMIZE: do this asynchronously?
+		if err := r.Get(ctx, client.ObjectKeyFromObject(&relay), &relay); err != nil {
+			return nil, fmt.Errorf("could not lookup circuitRelay: %w", err)
+		}
+		if err := relay.Status.AddrInfo.Parse(); err != nil {
+			log.Error(err, "could not parse AddrInfo. Information will not be included in config", "relay", relayName)
+			continue
+		}
+		addrInfo := relay.Status.AddrInfo.AddrInfo()
+		relayPeers = append(relayPeers, addrInfo)
+	}
+	return relayPeers, nil
+}
+
+// MaxIPSStorage Accepts a storage quantity and returns with a
+// calculated value to be used for setting the Max IPFS storage value
+// in Gigabytes.
+func MaxIPFSStorage(ipfsStorage resource.Quantity) resource.Quantity {
+	var storageMaxGB resource.Quantity
+	sizei64, _ := ipfsStorage.AsInt64()
+	sizeGB := sizei64 / 1024 / 1024 / 1024
+	var reducedSize int64
+	// if the disk is big, use a bigger percentage of it.
+	if sizeGB > 1024*8 {
+		reducedSize = sizeGB * 9 / 10
+	} else {
+		reducedSize = sizeGB * 8 / 10
+	}
+	storageMaxGB = *resource.NewQuantity(reducedSize, "G")
```

Comment:
I have the feeling that the suffix here should be `Gi`. Particularly given that we are then dividing this against KiBs.